web/app/Http/Controllers/PageController.php

```diff
@@ -4,19 +4,205 @@
+                $wikitext = true;
+                break;
+            case 'html':
+                $html = true;
+                break;
+            case 'none':
+                break;
+            default:
+                return null;
+        }
+
+        return [
+            'metadata' => $metadata,
+            'wikitext' => $wikitext,
+            'html' => $html,
+        ];
+    }
+
+    /**
+     * @return Page|Response
+     */
+    private function resolvePage(
+        string $path_type,
+        string $path,
+        bool $wikitext = false,
+        bool $html = false
+    ) {
+        $site = LegacyTools::getCurrentSite();
+        if ($site === null) {
+            return apierror(404, APIError::SITE_NOT_FOUND);
+        }
+
+        $site_id = $site->getSiteId();
+
+        $page = null;
+
+        if ($path_type === 'slug') {
+            $page = Page::findSlug($site_id, $path, $wikitext, $html);
+        } elseif ($path_type === 'id') {
+            $page = Page::findId($site_id, (int) $path, $wikitext, $html);
+        } else {
+            return apierror(400, APIError::INVALID_PAGE_PATH);
+        }
+
+        if ($page === null) {
+            return apierror(404, APIError::PAGE_NOT_FOUND);
+        }
+
+        return $page;
+    }
+
+    /**
+     * Gets a page.
+     * Endpoint: `GET:/page/{path_type}/{path}` | `pageGet`
+     */
+    public function pageGet(Request $request, string $path_type, string $path)
+    {
+        $output_type = $this->resolvePageType($request);
+        if ($output_type === null) {
+            return apierror(400, APIError::INVALID_PAGE_TYPE);
+        }
+
+        $page = $this->resolvePage(
+            $path_type,
+            $path,
+            $output_type['wikitext'],
+            $output_type['html'],
+        );
+
+        // api error was returned
+        if ($page instanceof Response) {
+            return $page;
+        }
+
+        $output = [];
+
+        if ($output_type['metadata']) {
+            $avatars = (bool) $request->query('avatars', false);
+
+            // TODO: get creator through earliest revision
+            // (unless we implement the authors field first)
+
+            $updater = User::findId($page->revision_user_id);
+
+            if ($updater !== null) {
+                $updater = $updater->toApiArray($avatars);
+            }
+
+            $output = array_merge($output, [
+                'id' => $page->id,
+                'slug' => $page->slug,
+                'category' => $page->page_category_id,
+                'parent' => null, // TODO
+                'children' => [], // TODO
+                'title' => $page->title,
+                'altTitle' => $page->alt_title,
+                'tags' => $page->tags,
+                'score' => 0, // TODO
+                'created' => $page->page_created_at,
+                'creator' => null, // TODO
+                'updated' => $page->lastUpdated(),
+                'updater' => $updater,
```

Comment:
This isn't necessarily true, `$updater` here is the last person to make a revision on the page, which may not be why the page was most recently updated. (Though in our current implementation this assumption holds afaik)

If you want "last revision time" then you want `revision_created_at`, not `page_updated_at`.